ui/src/app/utils/leo-runtime-initializer.tsx

```diff
@@ -343,9 +363,13 @@ export class LeoRuntimeInitializer {
           `Runtime ${this.currentRuntime.googleProject}/${this.currentRuntime.runtimeName}` +
           ` has reached an ERROR status`);
         await this.deleteRuntime();
-      } else if (this.reachedResolution()) {
-        // We've reached the goal - resolve the Promise.
-        return this.resolve(this.currentRuntime);
+      } else if (this.isRuntimeRunning()) {
```

Comment:
This breaks the functionality Eric added, since his `resolutionCondition` is `status !== Running`. I believe the correct logic here instead would be:

```
} else if (this.reachedResolution()) {
  ...
} else if (this.isRuntimeRunning() && this.pauseCount < this.maxPauseCount) {
  await this.pauseRuntime();
}
```

You should also adjust the resolutionCondition in your case to instead look for the Stopped status, otherwise it will just hang indefinitely.

Reply:
Sorry, on further reading, I realize it won't hang - but instead it will just throw once it hits the limit of 0 on maxResumeCount, which is similar to what we're doing for deletion. resolutionCondition is probably a cleaner way of handling this (per my other comment), both for this Stop functionality, and for the previous delete functionality.

Reply:
If I do this I'll need to move resolutionCondition up to the top of this set of conditionals or else it'll just thrash between stopping/starting forever. It's likely that it should be there anyways thoguh